src/rust/src/rustzcash.rs

```diff
@@ -282,39 +243,35 @@ pub extern "system" fn librustzcash_to_scalar(
     // Should be okay, because caller is responsible for ensuring
     // the pointer is a valid pointer to 32 bytes, and that is the
     // size of the representation
-    let scalar = <Bls12 as JubjubEngine>::Fs::to_uniform(unsafe { &(&*input)[..] }).into_repr();
-
-    let result = unsafe { &mut *result };
+    let scalar = Fr::from_bytes_wide(&mut unsafe { *input });
 
-    scalar
-        .write_le(&mut result[..])
-        .expect("length is 32 bytes");
+    *&mut unsafe { *result } = scalar.to_bytes();
```

Comment:
change this line to:
unsafe{*result = scalar.to_bytes()};
or result is not updated